tools.go

```diff
@@ -14,12 +14,35 @@ import (
 	"golang.org/x/time/rate"
 )
 
-// createTools creates the MCP tool definitions with appropriate rate limits
+const MCP_USER_AGENT = "axiom-mcp-server (MCP)"
+
+func createAxiomHTTPClient() *http.Client {
+	return &http.Client{
+		Transport: &mcpTransport{
+			base: http.DefaultTransport,
+		},
+	}
+}
+
+type mcpTransport struct {
+	base http.RoundTripper
+}
+
+func (t *mcpTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	// Clone the request to avoid modifying the original
+	reqCopy := req.Clone(req.Context())
+	reqCopy.Header.Set("User-Agent", MCP_USER_AGENT)
+	return t.base.RoundTrip(reqCopy)
+}
+
 func createTools(cfg config) ([]mcp.ToolDefinition, error) {
+	httpClient := createAxiomHTTPClient()
+
 	client, err := axiom.NewClient(
 		axiom.SetToken(cfg.token),
 		axiom.SetURL(cfg.url),
 		axiom.SetOrganizationID(cfg.orgID),
+		axiom.SetClient(httpClient),
```

Comment:
You can just use https://github.com/axiomhq/axiom-go/blob/83a8232dfb1a3856cad22ecc606b4de27d97b5a1/axiom/client_options.go#L71

Reply:
And please, set the user agent to: fmt.Sprintf(`mcp-server-axiom/%s`, Version)